R/preparedata.R

```diff
@@ -1395,12 +1393,36 @@ transformTable <- function(data,
     # hide.rows.threshold and row.names.to.remove refer to rows AFTER tranposing
     if (isTRUE(transpose))
     {
-        if (length(dim(data)) > 2)
+        if (!is.null(attr(data, "questions")) && length(dim(data)) == 2 && is.null(attr(data, "statistic")))
+        {
+            # 1-dimensional table with multiple statistics
+            is.qtable <- inherits(data, "QTable")
+            old.span <- attr(data, "span", exact = TRUE)
+            new.data <- array(data, dim = c(1, nrow(data), ncol(data)), dimnames = list("", rownames(data), colnames(data)))
+            data <- CopyAttributes(new.data, data)
+            if (is.qtable)
+                class(data) <- c(class(data), "QTable")
```

Comment:
Would this line make it 

old class  = c("QTable", "array")
new class = c("QTable", "array", "QTable")?
If so, why?

Reply:
Its because calling `CopyAttributes` drops the class